Components/WebService/Rest/RestConfigurableProducer.php

```diff
@@ -202,6 +202,12 @@ protected function request(ClientInterface $client, array $stepActionParams, arr
         $request = new Request($httpMethod, $resolvedURI, $requestHeaders);
         $response = null;
         try {
+
+            //setting the new error_handler
+            if(null !== $handlerStack = $client->getConfig('handler')) {
+                $handlerStack->push(Middleware::httpErrors(), 'http_errors_handler');
```

Comment:
We might need to move this step to the container or setup of the client, but not here. If you run the worker with a couple of messages going to the same producer, you end up stacking the same handler over and over and over. This is what I got after processing two messages that failed to be sent to the same target (`var_export()` of the handler stack):

```
array (
  0 => 
  array (
    0 => 
    Closure::__set_state(array(
    )),
    1 => 'http_errors',
  ),
  1 => 
  array (
    0 => 
    Closure::__set_state(array(
    )),
    1 => 'allow_redirects',
  ),
  2 => 
  array (
    0 => 
    Closure::__set_state(array(
    )),
    1 => 'cookies',
  ),
  3 => 
  array (
    0 => 
    Closure::__set_state(array(
    )),
    1 => 'prepare_body',
  ),
  4 => 
  array (
    0 => 
    Closure::__set_state(array(
    )),
    1 => 'http_errors_handler',
  ),
  5 => 
  array (
    0 => 
    Closure::__set_state(array(
    )),
    1 => 'http_errors_handler',
  ),
)
```

Notice the two `http_errors_handler` at the bottom of the array.

Reply:
Moved to UseGuzzeTrait